src/libs/deploy.js

```diff
@@ -0,0 +1,153 @@
+// Copyright © 2018 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+import * as minio from 'minio';
+import fs from 'fs';
+import cp from 'child_process';
+import util from 'util';
+import path from 'path';
+import shortid from 'shortid';
+import config from '../config';
+import { getObject } from './bucket';
+import { logger } from './logger';
+
+const exec = util.promisify(cp.exec);
+const writeFile = util.promisify(fs.writeFile);
+
+const bucket = config.get('minio:bucket');
+const client = new minio.Client({
+  endPoint: config.get('minio:endPoint'),
+  port: config.get('minio:port'),
+  secure: config.get('minio:secure'),
+  accessKey: config.get('minio:accessKey'),
+  secretKey: config.get('minio:secretKey'),
+  region: config.get('minio:region'),
+});
+
+/**
+ * Get the signed appliaction package
+ *
+ * @param {String} appPath The path to the signed app
+ */
+const fetchFileFromStorage = async (signedAppPath) => {
+  try {
+    const buffer = await getObject(client, bucket, signedAppPath);
+    if (!buffer) {
+      throw errorWithCode('Unable to fetch archive.', 500);
+    }
+    logger.info(`Get the bucket ${bucket} `);
+    return buffer;
+  } catch (error) {
+    const message = 'Unable to retrieve archive';
+    logger.error(`${message}, err = ${error.message}`);
+    throw error;
+  }
+};
+
+/**
+ * Google Play Store Deployment
+ *
+ * @param {*} signedAppPath The path to the signed app
+ * @returns The status of the deployment
+ */
+// eslint-disable-next-line import/prefer-default-export
+export const deployApk = async (signedAppPath) => {
+  const signedAPK = await fetchFileFromStorage(signedAppPath);
```

Comment:
This await should be in a try block.

Reply:
I think it should still be in a try catch. You can just re-throw the same error if needed. In any case when a promise is returned it can be resolved (nothing to do) or rejected (needs to be caught). What' I'd probably do is throw an error with a meaningful meesage like `throw new Error(`Unable to fetch file ${NAME} from storage`);